lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -53,9 +53,19 @@ module.exports = {
   async uploadZipFile(artifactFilePath) {
     const fileName = artifactFilePath.split(path.sep).pop();
 
-    // TODO refactor to be async (use util function to compute checksum async)
-    const data = fs.readFileSync(artifactFilePath);
-    const fileHash = crypto.createHash('sha256').update(data).digest('base64');
+    const fileHasher = crypto.createHash('sha256');
+    const fileHash = await new Promise((resolve, reject) => {
```

Comment:
Instead of introducing such logic here I think better option would be to reuse an existing utility function: https://github.com/serverless/serverless/blob/master/lib/plugins/aws/package/lib/getHashForFilePath.js

Reply:
I can easily switch it to that but fixing the tests is quite onerous since they depend heavily on the actual implementation rather than the behavior of the functions under test.